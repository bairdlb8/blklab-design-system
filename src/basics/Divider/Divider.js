import React from 'react'
import PropTypes from 'prop-types'

import './Divider.scss'

const Divider = ({ className = '', margin = 4 }) => {
  return <div className={`ds-divider display-flow-root my-${margin}`}></div>
}

Divider.propTypes = {}

export default Divider
