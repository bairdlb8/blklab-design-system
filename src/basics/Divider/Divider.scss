.ds-divider {
  --divider-thickness: 1px;
  height: var(--divider-thickness);
  background-color: var(--ds-color-gray-light);
}
