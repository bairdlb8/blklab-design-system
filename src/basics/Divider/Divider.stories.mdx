import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Divider from './Divider'

<Meta title="Basics / Divider" component={ Divider } />

# Divider

## Add some text here

export const Template = (args) => {
  return <Divider {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={ {} }>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={ Divider } />
