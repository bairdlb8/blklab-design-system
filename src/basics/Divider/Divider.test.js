import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { toHaveNoViolations } from 'jest-axe'
import Divider from './Divider'

expect.extend(toHaveNoViolations)

describe('<Divider />', () => {

  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Divider />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Divider />)

    expect(asFragment(<Divider />)).toMatchSnapshot()
  })

})
