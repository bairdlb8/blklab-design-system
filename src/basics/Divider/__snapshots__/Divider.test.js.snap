// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Divider
  </div>
</DocumentFragment>
`;
